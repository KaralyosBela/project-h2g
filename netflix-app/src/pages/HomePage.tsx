import { useEffect } from "react";
import { Banner } from "../components/Banner";
import { FilterBar } from "../components/FilterBar";
import { Footer } from "../components/Footer";
import { MovieList } from "../components/MovieList";
import { Layout } from "../ui/Layout";
import { useDispatch } from "react-redux";
import { AppDispatch } from "../app/store";
import { getMovies } from "../features/moviesSlice";
import { MovieBanner } from "../components/MovieBanner";
import { useMovies } from "../features/movies.hook";
import { useAppSelector } from "../app/hooks";
//Action gyakorlas
import { customAction } from "../features/actions";

export const HomePage: React.FC = () => {
  const movieSelected = useAppSelector((state) => state.movies.movie);
  const bannerVisible = useAppSelector((state) => state.movies.bannerVisible);
  const dispatch = useDispatch<AppDispatch>();
  const movies = useMovies();

  useEffect(() => {
    dispatch(getMovies());
    //Action gyakorlas
    dispatch(customAction("Hello Action"));
  },[]);
  
  return (
    <div>
      <Layout>
        {bannerVisible && <MovieBanner movie={movieSelected} />}
        {!bannerVisible && <Banner />}
        <FilterBar movieCount={movies.length} />
        {movies.length > 0 && <MovieList moviesList={movies} />}
        <Footer />
      </Layout>
    </div>
  );
};
