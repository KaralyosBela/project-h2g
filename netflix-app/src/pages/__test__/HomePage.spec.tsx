import { render, screen } from "@testing-library/react";
import { HomePage } from "../HomePage";
import { IMovie } from "../../interfaces/movies.interface";

const mockDispatch = jest.fn();
const mockUseDispatch = jest.fn(() => mockDispatch); //itt is megtudom hívni
// const dispatch = useDispatch<AppDispatch>(); kb ami itt történik
const mockUseAppSelector = jest.fn();
const mockUseMovies = jest.fn();

// jest.mock("../../components/MovieList", () => ({
//   MovieList: () => <div>MovieList</div>,
// }));

jest
  .mock("../../app/hooks.ts", () => ({
    useAppSelector: () => mockUseAppSelector(),
  }))
  .mock("react-redux", () => ({
    useDispatch: () => mockUseDispatch, //itt eddig megvolt hívva a mockUseDispatch()
  }))
  .mock("../../features/movies.hook.ts", () => ({
    useMovies: () => mockUseMovies(),
  }));

describe("Homepage", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render component without the movie banner, with the default banner", () => {
    const movies: IMovie[] = [
      {
        id: "1",
        title: "Cim1",
        tagline: "alcim1",
        vote_average: 4,
        vote_count: 4,
        budget: 4,
        revenue: 4,
        release_date: "2019 01 02",
        genres: ["asd234asd", "as324d"],
        poster_path: "path:://asdasdasd",
        runtime: 4,
        overview: "the best movie ever",
      },
      {
        id: "2",
        title: "cim2",
        tagline: "alcim2",
        vote_average: 4,
        vote_count: 4,
        budget: 4,
        revenue: 4,
        release_date: "2011 02 02",
        genres: ["sdfsdfd", "asddfsd"],
        poster_path: "path://asdasd34234",
        runtime: 4,
        overview: "the best movie ever 2",
      },
    ];
    const movieSelected = {
      id: "string",
      title: "string",
      tagline: "string",
      vote_average: 4,
      vote_count: 4,
      budget: 4,
      revenue: 4,
      release_date: "string",
      genres: ["asd", "asd"],
      poster_path: "string",
      runtime: 4,
      overview: "string",
    };
    const bannerVisible = false;

    mockUseAppSelector.mockReturnValueOnce(movieSelected);
    mockUseAppSelector.mockReturnValueOnce(bannerVisible);
    mockUseMovies.mockReturnValue(movies);

    const { container } = render(<HomePage />);

    expect(container).toMatchSnapshot();
    expect(mockUseDispatch.mock.calls.length).toEqual(1);
    // screen.debug();
  });

  it("should render component with the movie banner, without the default banner", () => {
    const movies: IMovie[] = [
      {
        id: "1",
        title: "Cim1",
        tagline: "alcim1",
        vote_average: 4,
        vote_count: 4,
        budget: 4,
        revenue: 4,
        release_date: "2019 01 02",
        genres: ["asd234asd", "as324d"],
        poster_path: "path:://asdasdasd",
        runtime: 4,
        overview: "the best movie ever",
      },
      {
        id: "2",
        title: "cim2",
        tagline: "alcim2",
        vote_average: 4,
        vote_count: 4,
        budget: 4,
        revenue: 4,
        release_date: "2011 02 02",
        genres: ["sdfsdfd", "asddfsd"],
        poster_path: "path://asdasd34234",
        runtime: 4,
        overview: "the best movie ever 2",
      },
    ];
    const movieSelected = {
      id: "string",
      title: "string",
      tagline: "string",
      vote_average: 4,
      vote_count: 4,
      budget: 4,
      revenue: 4,
      release_date: "string",
      genres: ["asd", "asd"],
      poster_path: "string",
      runtime: 4,
      overview: "string",
    };
    const bannerVisible = true;

    mockUseAppSelector.mockReturnValueOnce(movieSelected);
    mockUseAppSelector.mockReturnValueOnce(bannerVisible);
    mockUseMovies.mockReturnValue(movies);

    const { container } = render(<HomePage />);

    expect(container).toMatchSnapshot();
    expect(mockUseDispatch.mock.calls.length).toEqual(1);
    // screen.debug();
  });

  it("should render component without the movieslist component", () => {
    const movies: IMovie[] = [];
    const movieSelected = {
      id: "string",
      title: "string",
      tagline: "string",
      vote_average: 4,
      vote_count: 4,
      budget: 4,
      revenue: 4,
      release_date: "string",
      genres: ["asd", "asd"],
      poster_path: "string",
      runtime: 4,
      overview: "string",
    };
    const bannerVisible = true;

    mockUseAppSelector.mockReturnValueOnce(movieSelected);
    mockUseAppSelector.mockReturnValueOnce(bannerVisible);
    mockUseMovies.mockReturnValue(movies);

    const { container } = render(<HomePage />);

    expect(container).toMatchSnapshot();
    expect(mockUseDispatch.mock.calls.length).toEqual(1);
    // screen.debug();
  });
});
