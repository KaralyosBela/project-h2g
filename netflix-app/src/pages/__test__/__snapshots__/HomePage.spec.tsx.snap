// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render component with the movie banner, without the default banner 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="movieBanner"
      >
        <div
          class="titleBar"
        >
          <h3
            class="pageTitle"
          >
            netflixRoulette
          </h3>
          <div
            class="searchIcon"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
        </div>
        <div
          class="infoPart"
        >
          <img
            alt="alt"
            class="image"
            src="string"
          />
          <div
            class="movieInfo"
          >
            <div
              class="firstRow"
            >
              <h1
                class="title"
              >
                string
              </h1>
              <span
                class="vote_average"
              >
                4
              </span>
            </div>
            <p
              class="genre"
            >
              asd, asd
            </p>
            <div
              class="secondRow"
            >
              <h3
                class="release_date"
              >
                string
              </h3>
              <h3
                class="runtime"
              >
                4
                 min
              </h3>
            </div>
            <div
              class="thirdRow"
            >
              <p
                class="overview"
              >
                string
              </p>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="filterBar"
        >
          <div
            class="categories"
          >
            <button
              class="active"
            >
              All
            </button>
            <button
              class=""
            >
              Action
            </button>
            <button
              class=""
            >
              Comedy
            </button>
            <button
              class=""
            >
              Horror
            </button>
            <button
              class=""
            >
              Crime
            </button>
          </div>
          <div
            class="sortSection"
          >
            <div>
              Sort by
            </div>
            <select>
              <option
                value="releaseDate_ascending"
              >
                Release date asc
              </option>
              <option
                value="releaseDate_descending"
              >
                Release date desc
              </option>
              <option
                value="length_ascending"
              >
                Length asc
              </option>
              <option
                value="length_descending"
              >
                Length desc
              </option>
              <option
                value="rating_ascending"
              >
                Rating asc
              </option>
              <option
                value="rating_descending"
              >
                Rating desc
              </option>
            </select>
          </div>
        </div>
        <div
          class="movieCounter"
        >
          <span>
            2
          </span>
           movies found
        </div>
      </div>
      <div
        class="mainSection"
      >
        <div
          class="movieContainer"
        >
          <div>
            <div
              class="card"
              data-testid="card"
            >
              <div
                class="circle"
                data-testid="circle"
              >
                <div
                  class="firstDot"
                />
                <div
                  class="secondDot"
                />
                <div
                  class="thirdDot"
                />
              </div>
              <img
                alt="Not found"
                class="image"
                src="path:://asdasdasd"
              />
              <div
                class="info"
              >
                <div
                  class="title"
                >
                  Cim1
                </div>
                <div
                  class="year"
                >
                  2019 01 02
                </div>
              </div>
              <div
                class="genre"
              >
                asd234asd, as324d
              </div>
            </div>
          </div>
          <div>
            <div
              class="card"
              data-testid="card"
            >
              <div
                class="circle"
                data-testid="circle"
              >
                <div
                  class="firstDot"
                />
                <div
                  class="secondDot"
                />
                <div
                  class="thirdDot"
                />
              </div>
              <img
                alt="Not found"
                class="image"
                src="path://asdasd34234"
              />
              <div
                class="info"
              >
                <div
                  class="title"
                >
                  cim2
                </div>
                <div
                  class="year"
                >
                  2011 02 02
                </div>
              </div>
              <div
                class="genre"
              >
                sdfsdfd, asddfsd
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
        data-testid="footer"
      >
        <h4
          data-testid="header"
        >
          <span
            data-testid="span"
          >
            netflix
          </span>
          roulette
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`Homepage should render component without the movie banner, with the default banner 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="picture"
      >
        <div
          class="banner"
        >
          <div
            class="upperBar"
          >
            <div>
              <h4
                class="upperBarTitle"
              >
                <span>
                  netflix
                </span>
                Roulette
              </h4>
            </div>
            <button
              class="addButton"
            >
              ADD MOVIE
            </button>
          </div>
          <div
            class="title"
          >
            <h1>
              FIND YOUR MOVIE
            </h1>
          </div>
          <div
            class="searchBar"
          >
            <input
              class="searchInput"
              placeholder="What do you want to watch?"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="filterBar"
        >
          <div
            class="categories"
          >
            <button
              class="active"
            >
              All
            </button>
            <button
              class=""
            >
              Action
            </button>
            <button
              class=""
            >
              Comedy
            </button>
            <button
              class=""
            >
              Horror
            </button>
            <button
              class=""
            >
              Crime
            </button>
          </div>
          <div
            class="sortSection"
          >
            <div>
              Sort by
            </div>
            <select>
              <option
                value="releaseDate_ascending"
              >
                Release date asc
              </option>
              <option
                value="releaseDate_descending"
              >
                Release date desc
              </option>
              <option
                value="length_ascending"
              >
                Length asc
              </option>
              <option
                value="length_descending"
              >
                Length desc
              </option>
              <option
                value="rating_ascending"
              >
                Rating asc
              </option>
              <option
                value="rating_descending"
              >
                Rating desc
              </option>
            </select>
          </div>
        </div>
        <div
          class="movieCounter"
        >
          <span>
            2
          </span>
           movies found
        </div>
      </div>
      <div
        class="mainSection"
      >
        <div
          class="movieContainer"
        >
          <div>
            <div
              class="card"
              data-testid="card"
            >
              <div
                class="circle"
                data-testid="circle"
              >
                <div
                  class="firstDot"
                />
                <div
                  class="secondDot"
                />
                <div
                  class="thirdDot"
                />
              </div>
              <img
                alt="Not found"
                class="image"
                src="path:://asdasdasd"
              />
              <div
                class="info"
              >
                <div
                  class="title"
                >
                  Cim1
                </div>
                <div
                  class="year"
                >
                  2019 01 02
                </div>
              </div>
              <div
                class="genre"
              >
                asd234asd, as324d
              </div>
            </div>
          </div>
          <div>
            <div
              class="card"
              data-testid="card"
            >
              <div
                class="circle"
                data-testid="circle"
              >
                <div
                  class="firstDot"
                />
                <div
                  class="secondDot"
                />
                <div
                  class="thirdDot"
                />
              </div>
              <img
                alt="Not found"
                class="image"
                src="path://asdasd34234"
              />
              <div
                class="info"
              >
                <div
                  class="title"
                >
                  cim2
                </div>
                <div
                  class="year"
                >
                  2011 02 02
                </div>
              </div>
              <div
                class="genre"
              >
                sdfsdfd, asddfsd
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
        data-testid="footer"
      >
        <h4
          data-testid="header"
        >
          <span
            data-testid="span"
          >
            netflix
          </span>
          roulette
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`Homepage should render component without the movieslist component 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="movieBanner"
      >
        <div
          class="titleBar"
        >
          <h3
            class="pageTitle"
          >
            netflixRoulette
          </h3>
          <div
            class="searchIcon"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
        </div>
        <div
          class="infoPart"
        >
          <img
            alt="alt"
            class="image"
            src="string"
          />
          <div
            class="movieInfo"
          >
            <div
              class="firstRow"
            >
              <h1
                class="title"
              >
                string
              </h1>
              <span
                class="vote_average"
              >
                4
              </span>
            </div>
            <p
              class="genre"
            >
              asd, asd
            </p>
            <div
              class="secondRow"
            >
              <h3
                class="release_date"
              >
                string
              </h3>
              <h3
                class="runtime"
              >
                4
                 min
              </h3>
            </div>
            <div
              class="thirdRow"
            >
              <p
                class="overview"
              >
                string
              </p>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="filterBar"
        >
          <div
            class="categories"
          >
            <button
              class="active"
            >
              All
            </button>
            <button
              class=""
            >
              Action
            </button>
            <button
              class=""
            >
              Comedy
            </button>
            <button
              class=""
            >
              Horror
            </button>
            <button
              class=""
            >
              Crime
            </button>
          </div>
          <div
            class="sortSection"
          >
            <div>
              Sort by
            </div>
            <select>
              <option
                value="releaseDate_ascending"
              >
                Release date asc
              </option>
              <option
                value="releaseDate_descending"
              >
                Release date desc
              </option>
              <option
                value="length_ascending"
              >
                Length asc
              </option>
              <option
                value="length_descending"
              >
                Length desc
              </option>
              <option
                value="rating_ascending"
              >
                Rating asc
              </option>
              <option
                value="rating_descending"
              >
                Rating desc
              </option>
            </select>
          </div>
        </div>
        <div
          class="movieCounter"
        >
          <span>
            0
          </span>
           movies found
        </div>
      </div>
      <div
        class="footer"
        data-testid="footer"
      >
        <h4
          data-testid="header"
        >
          <span
            data-testid="span"
          >
            netflix
          </span>
          roulette
        </h4>
      </div>
    </div>
  </div>
</div>
`;
